vpr/src/route/connection_router.cpp

```diff
@@ -477,27 +535,55 @@ void ConnectionRouter<Heap>::timing_driven_expand_neighbour(t_heap* current,
     VTR_LOGV_DEBUG(router_debug_, "      Expanding node %d edge %zu -> %d\n",
                    from_node, size_t(from_edge), to_node_int);
 
-    timing_driven_add_to_heap(cost_params,
-                              current,
-                              from_node,
-                              to_node_int,
-                              from_edge,
-                              target_node);
+    // Check if the node exists in the route tree, needed for RCV code
+
+    bool node_exists = false;
+    bool run_rcv = cost_params.delay_budget && cost_params.delay_budget->routing_budgets_algorithm == YOYO;
+    if (run_rcv) {
+        node_exists = node_exists_in_route_tree(current,
+                                                to_node,
+                                                route_tree_nodes);
+    }
+
+    if (!node_exists || !run_rcv) {
+        timing_driven_add_to_heap(cost_params,
+                                  current,
+                                  from_node,
+                                  to_node_int,
+                                  from_edge,
+                                  target_node,
+                                  run_rcv);
+    }
 }
 
 //Add to_node to the heap, and also add any nodes which are connected by non-configurable edges
 template<typename Heap>
 void ConnectionRouter<Heap>::timing_driven_add_to_heap(const t_conn_cost_params cost_params,
-                                                       const t_heap* current,
```

Comment:
I don't believe you need to mutate "current" here, so why did you drop the "const"?

Reply:
Good point, I think I dropped it a long time ago for debugging purposes and didn't add it back in. I'll add this back in.

Reply:
Still outstanding

Reply:
Sorry yes, I'm getting to this incrementally! You're right that there's still these issues outstanding